internal/backoff/backoff_test.go

```diff
@@ -472,74 +237,358 @@ func Test_backoff_Close(t *testing.T) {
 		backoffTimeLimit      time.Duration
 		errLog                bool
 	}
-	type want struct{}
+	type want struct {
+		wantRes interface{}
+		err     error
+	}
 	type test struct {
 		name       string
+		args       args
 		fields     fields
 		want       want
-		checkFunc  func(want) error
-		beforeFunc func()
-		afterFunc  func()
+		checkFunc  func(want, interface{}, error) error
+		beforeFunc func(args)
+		afterFunc  func(args)
 	}
-	defaultCheckFunc := func(w want) error {
+	defaultCheckFunc := func(w want, gotRes interface{}, err error) error {
+		if !errors.Is(err, w.err) {
+			return errors.Errorf("got_error: \"%#v\",\n\t\t\t\twant: \"%#v\"", err, w.err)
+		}
+		if !reflect.DeepEqual(gotRes, w.wantRes) {
+			return errors.Errorf("got: \"%#v\",\n\t\t\t\twant: \"%#v\"", gotRes, w.wantRes)
+		}
 		return nil
 	}
 	tests := []test{
-		// TODO test cases
-		/*
-		   {
-		       name: "test_case_1",
-		       fields: fields {
-		           wg: sync.WaitGroup{},
-		           backoffFactor: 0,
-		           initialDuration: 0,
-		           jittedInitialDuration: 0,
-		           jitterLimit: 0,
-		           durationLimit: 0,
-		           maxDuration: 0,
-		           maxRetryCount: 0,
-		           backoffTimeLimit: nil,
-		           errLog: false,
-		       },
-		       want: want{},
-		       checkFunc: defaultCheckFunc,
-		   },
-		*/
-
-		// TODO test cases
-		/*
-		   func() test {
-		       return test {
-		           name: "test_case_2",
-		           fields: fields {
-		           wg: sync.WaitGroup{},
-		           backoffFactor: 0,
-		           initialDuration: 0,
-		           jittedInitialDuration: 0,
-		           jitterLimit: 0,
-		           durationLimit: 0,
-		           maxDuration: 0,
-		           maxRetryCount: 0,
-		           backoffTimeLimit: nil,
-		           errLog: false,
-		           },
-		           want: want{},
-		           checkFunc: defaultCheckFunc,
-		       }
-		   }(),
-		*/
+		func() test {
+			ctx := context.Background()
+			err := errors.New("error is occurred")
+			f := func(context.Context) (interface{}, bool, error) {
+				return nil, false, err
+			}
+			return test{
+				name: "return nil response and nil error when function returns (nil, false, error)",
```

Comment:
which one do you think is better?
```suggestion
				name: "return nil response and nil error when function is not retriable",
```

Reply:
I think `retriable` has some meaning in this case.

Reply:
how about this?
Why I comment this is because I think that it is hard to understand the purpose when reading the test case name.
```suggestion
				name: "return nil response and error when function return not retriable",
```

Reply:
How is this?
```suggestion
				name: "return nil response and nil error when function returns (nil, false, error) and not retriable",
```

Reply:
btw it should be 
```suggestion
				name: "return nil response and error when function returns (nil, false, error) and not retriable",
```

Reply:
please also fix other test case name